py/desiutil/redirect.py

```diff
@@ -0,0 +1,247 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+"""
+===================
+desiutil.redirect
+===================
+
+Utilities for redirecting stdout / stderr to files.
+
+"""
+
+import os
+import sys
+import time
+import io
+import traceback
+import logging
+import ctypes
+
+from contextlib import contextmanager
+
+from .log import get_logger, _desiutil_log_root
+
+
+# C file descriptors for stderr and stdout, used in redirection
+# context manager.
+
+libc = ctypes.CDLL(None)
```

Comment:
Why does this code need to be called at the module level?  Calling functions at import time has been the source of much pain in the past.

Reply:
That block of code gets a handle to the system `libc` and the low-level `STDOUT` and `STDERR`.  We definitely only want to do that once, so it seemed fine to do that on module import.  Of course I can make a function that sets those module level variables if they are not set and returns them.  However, for the record this code is copied out of desispec where it has been working well for years.

Reply:
This is now wrapped in a function